src/pykeen/nn/emb.py

```diff
@@ -108,13 +108,34 @@ def forward(
     ) -> torch.FloatTensor:
         """Get representations for indices.
 
+        .. note ::
+            this method is implemented in subclasses. Prefer using `get` instead, which optimizes for duplicate indices.
+
+        :param indices: shape: s
+            The indices, or None. If None, this is interpreted as ``torch.arange(self.max_id)`` (although implemented
+            more efficiently).
+
+        :return: shape: (``*s``, ``*self.shape``)
+            The representations.
+        """
```

Comment:
This needs a marking with an abstract class or raise not implemented

Reply:
It is already an `abstractmethod` 🤔

Reply:
Sorry I just didn’t see it in the diff